iac-extensions/docker/src/main/java/org/sonar/iac/docker/checks/PackageInstallationCacheCheck.java

```diff
@@ -148,6 +176,46 @@ private static List<CommandDetector.Command> detectRemoveCacheCommands(
       .toList();
   }
 
+  /* Compute a set of commands for which their cache path is present is the provided list. */
+  private static Set<String> computeCachedCommands(List<String> cachedPaths) {
+    return cachedPaths.stream()
+      .flatMap(cachedPath -> CACHE_TO_COMMANDS.getOrDefault(cachedPath, Collections.emptySet()).stream())
+      .collect(Collectors.toSet());
```

Comment:
I find the stream logic here a bit hard to read.
IMO this would be more readable
```
Set<String> result = new HashSet<>();
for (String cachedPath : cachedPaths) {
  if(Map.contains(cachedPath)) {
    result.add(Map.get(cachedPath))
  }
}
return result
```

WDYT?